.github/workflows/ci.yml

```diff
@@ -135,3 +135,68 @@ jobs:
       - uses: actions/checkout@v4
       - name: Build Docker Image
         run: docker build --target web -t ons .
+
+  functional-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        browser:
+          - chromium
+    needs:
+      - lint
+      - lint-front-end
+      - compile_static
+
+    env:
+      DJANGO_SETTINGS_MODULE: cms.settings.functional_test
+      SECRET_KEY: fake_secret_key_to_run_tests # pragma: allowlist secret
+      PLAYWRIGHT_BROWSER: ${{ matrix.browser }}
+      PLAYWRIGHT_TRACES_DIR: playwright_traces
+      PLAYWRIGHT_TRACE: 'true'
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Poetry
+        run: pipx install poetry==${{ env.POETRY_VERSION }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version-file: .python-version
+          cache: poetry
+
+      - name: Install dependencies
+        run: make install-dev
+
+      - name: Playwright Install
+        run: poetry run python -m playwright install --with-deps ${{ matrix.browser }}
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: static
+          path: cms/static_compiled/
+
+      - name: Docker Compose Up
+        run: make functional-tests-up
+
+      - name: Fetch Design System templates
+        run: make load-design-system-templates
+
+      - name: Collect Static Files
+        run: poetry run ./manage.py collectstatic --verbosity 0 --noinput --clear
+
+      - name: Prepare Database
+        run: poetry run ./manage.py migrate --noinput
```

Comment:
so this is failing for me locally

<details>
<summary>shell outputs</summary>

```shell
~ ❯❯❯ PLAYWRIGHT_HEADLESS=False PLAYWRIGHT_SLOW_MO=1000 make functional-tests
docker compose -f functional_tests/docker-compose.yml up -d
[+] Running 4/4
 ✔ Network functional_tests_default    Created                                                                            0.1s
 ✔ Volume "functional_tests_pgdata"    Created                                                                            0.0s
 ✔ Container functional_tests-db-1     Started                                                                            0.4s
 ✔ Container functional_tests-redis-1  Started                                                                            0.4s
poetry run behave functional_tests
/Users/dan/Sites/ons/.venv/lib/python3.12/site-packages/behave/matchers.py:267: SyntaxWarning: invalid escape sequence '\d'
  """Registers a custom type that will be available to "parse"
/Users/dan/Sites/ons/.venv/lib/python3.12/site-packages/behave/formatter/ansi_escapes.py:57: SyntaxWarning: invalid escape sequence '\['
  _ANSI_ESCAPE_PATTERN = re.compile(u"\x1b\[\d+[mA]", re.UNICODE)
Creating test database for alias 'default'...
/Users/dan/Sites/ons/.venv/lib/python3.12/site-packages/django/db/backends/postgresql/base.py:512: RuntimeWarning: Normally Django will use a connection to the 'postgres' database to avoid running initialization queries against the production database when it's not needed (for example, when running tests). Django was unable to create a connection to the 'postgres' database and will use the first PostgreSQL database instead.
  warnings.warn(
HOOK-ERROR in before_all: OperationalError: connection failed: connection to server at "127.0.0.1", port 15432 failed: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
HOOK-ERROR in after_all: AttributeError: 'Context' object has no attribute 'playwright_trace'

ABORTED: By user.
0 features passed, 0 failed, 0 skipped, 4 untested
0 scenarios passed, 0 failed, 0 skipped, 4 untested
0 steps passed, 0 failed, 0 skipped, 0 undefined, 22 untested
Took 0m0.000s
make: *** [functional-tests-run] Error 1
(.venv)
# dan at tbx-m1p-zero.local in ~/Sites/ons on git:playwright-functional-tests-CMS-198 ● [15:14:07]
~ ❯❯❯ PLAYWRIGHT_HEADLESS=False PLAYWRIGHT_SLOW_MO=1000 make functional-tests
docker compose -f functional_tests/docker-compose.yml up -d
[+] Running 2/0
 ✔ Container functional_tests-redis-1  Running                                                                            0.0s
 ✔ Container functional_tests-db-1     Running                                                                            0.0s
poetry run behave functional_tests
Creating test database for alias 'default'...
HOOK-ERROR in before_all: ProgrammingError: relation "auth_group" does not exist
LINE 1: ...URSOR WITH HOLD FOR SELECT "auth_group"."id" FROM "auth_grou...
                                                             ^
HOOK-ERROR in after_all: AttributeError: 'Context' object has no attribute 'playwright_trace'

ABORTED: By user.
0 features passed, 0 failed, 0 skipped, 4 untested
0 scenarios passed, 0 failed, 0 skipped, 4 untested
0 steps passed, 0 failed, 0 skipped, 0 undefined, 22 untested
Took 0m0.000s
make: *** [functional-tests-run] Error 1
(.venv)
```

</details>

and if I change `functional-tests-run` to

```makefile
functional-tests-run:  ## Run the functional tests
	poetry run ./manage.py collectstatic --verbosity 0 --noinput --clear
	poetry run ./manage.py migrate --noinput
	poetry run behave functional_tests
```

it fails with no connection to the DB as I expect I'd need to massage `.env` to point to the right funcational db and redis

It may be worth adding a bash script that wraps it all (i.e. sets the right env vars inline before calling `behave functional_tests`

Reply:
yep, changing to
```
functional-tests-run:  ## Run the functional tests
	poetry run ./manage.py collectstatic --verbosity 0 --noinput --clear
	DATABASE_URL=postgres://ons:pass@localhost:15432/postgres REDIS_URL=redis://localhost:16379 poetry run ./manage.py migrate --noinput
	DATABASE_URL=postgres://ons:pass@localhost:15432/postgres REDIS_URL=redis://localhost:16379 poetry run behave functional_tests
```

makes it work

Reply:
Yes, I realised after splitting out the dev app from the docker compose that the test DB relies on having the main default DB structure there to copy, rather than creating it entirely from scratch itself.

You shouldn't need to pass the config through to the run tests command though, as it should be hardcoded to use the correct settings module in the behave `environment.py`, I'll investigate today and get it more robust in local environments.

Reply:
this works now. thanks!